rasa/core/featurizers/tracker_featurizers.py

```diff
@@ -0,0 +1,356 @@
+import jsonpickle
+import logging
+import os
+from tqdm import tqdm
+from typing import Tuple, List, Optional, Dict, Text, Deque, FrozenSet
+import numpy as np
+
+import rasa.utils.io
+from rasa.core.featurizers.single_state_featurizer import SingleStateFeaturizer
+from rasa.core.domain import Domain, State
+from rasa.core.events import ActionExecuted
+from rasa.core.trackers import DialogueStateTracker, FrozenState
+from rasa.utils.common import is_logging_disabled
+from rasa.utils.features import Features
+from rasa.core.interpreter import NaturalLanguageInterpreter
+from rasa.core.constants import USER
+from rasa.nlu.constants import TEXT
+
+logger = logging.getLogger(__name__)
+
+
+class TrackerFeaturizer:
+    """Base class for actual tracker featurizers."""
+
+    def __init__(
+        self, state_featurizer: Optional[SingleStateFeaturizer] = None
+    ) -> None:
+
+        self.state_featurizer = state_featurizer
+
+    @staticmethod
+    def _unfreeze_states(states: Deque[FrozenState]) -> List[State]:
+        return [
+            {key: dict(value) for key, value in dict(state).items()} for state in states
+        ]
+
+    def _create_states(
+        self, tracker: DialogueStateTracker, domain: Domain
+    ) -> List[State]:
+        """Create states: a list of dictionaries."""
```

Comment:
please use either a full docstring or none (as it's private). We need to stay consistent with our docstrings.